qa/rpc-tests/test_framework/util.py

```diff
@@ -287,6 +291,18 @@ def sync_mempools(rpc_connections, wait=1,verbose=1):
             logging.info("sync mempool: " + str(pool_len))
         if num_match == len(rpc_connections):
             break
+
+        if ((count&3)==0):
+            # mempools may never sync fully because there is a bloom filter that can block some INVs
+            # so randomly push all txs from one node to another until the sync happens
```

Comment:
Not sure I really like the idea of this section. What I'm afraid of is that we'll end up masking issues with tx propagation, such as with the fast filter and an issue could slip past us into mainnet at some point.  The inv false positive should be 1 in a million, at least that's the false positive rate of the old style rolling bloom filters, so it doesn't seem like something we need to fix.

Reply:
We have timeouts and tests interrupted due to ctrl-c all the time.  Maybe it's not travis fault.  Suspiciously, these problems happen often in wallet backup.py, sendrawtx.py and others which whenever I look have mempool sync.  

I think the problem is that the chance isn't 1 in a million if the bloom is set for that.  It's the birthday paradox.  And this it just for 1 call.  How many sync calls per test run?  And then travis has 9 platforms it runs on.

We need the mempool sync to actually work 100% reliably so we dont waste so much time on weird failures, and test the filter performance separately.  I have a test for fast filter collision but right now it's set much higher than 1 in a million.

But I still need to work on this.  It doesn't solve the filters in AlreadyReceivedTx so mempool sync still isn't 100%.

Reply:
I think this one I can't agree with.  Not until we've moved away from Travis to GitLab and see where the chips fall there.  As far as I've seen , yes we've always had some spurious issue in wallet and wallet_backup but mostly the tests run well with the exception lately of Travis timing out the build process.

Reply:
Ok. Fair enough.  Prior to the parallel mempool admission PR, I fixed a bunch of reliability issues in the qa tests so we really haven't had the chance to run with those fixes and without the k=1 fastfilter (that has more collisions).  

Right now if I run the full suite on my machine a bunch of times and leave, I'm sure to be hung with unsynchronized mempools when I get back.  So I'm going to finish this PR up as a "demonstrator" (there's not much left to do) but let's not merge it.  

I'll then fix fastfilter and let's see how well things work.